internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -827,6 +849,19 @@ func dataSourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{}
 	return nil
 }
 
+func flattenKubernetesClusterDataSourceKeyVaultKms(input *managedclusters.AzureKeyVaultKms) []interface{} {
+	azureKeyVaultKms := make([]interface{}, 0)
+
+	if input != nil && *input.Enabled {
+		azureKeyVaultKms = append(azureKeyVaultKms, map[string]interface{}{
+			"key_vault_key_id":         input.KeyId,
+			"key_vault_network_access": input.KeyVaultNetworkAccess,
+		})
+	}
```

Comment:
We need to nil check these
```suggestion
	if input != nil && input.Enabled != nil && *input.Enabled {
		keyId := ""
		if v := input.KeyId; v != nil {
			keyId = *v
		}

		networkAccess := ""
		if v := input.KeyVaultNetworkAccess; v != nil {
			networkAccess = string(*v)
		}

		azureKeyVaultKms = append(azureKeyVaultKms, map[string]interface{}{
			"key_vault_key_id":         keyId,
			"key_vault_network_access": networkAccess,
		})
	}
```

Reply:
Makes sense :). Done. I'll test it again now.